### PR TITLE
refactor: delete cached configFilename

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- delete cached config filename if it doesn't exist

## Context
- https://github.com/renovatebot/renovate/pull/20733
- In the linked PR I have introduced functionality that detects whether repo is onboarded/not using cache. But we do this detection before using API to search for config-file in repo, thus this part of the code won't be executed. And in the case where repo is not onboarded and onboarding cache is valid we skip the part where we delete the cached configFilename
https://github.com/renovatebot/renovate/blob/c41cfdbef6e944ad374ae506d469c587d3567947/lib/workers/repository/onboarding/branch/check.ts#L64-L89

This causes error in the function below when we try to detectRepoFileConfig as we look for a config-file using api but won't find it.
This causes `unknown error` or `file not found` error when running on a real repo. So I am adding a try catch to handle the error better.
https://github.com/renovatebot/renovate/blob/e3c92d2f0139be2301a80bbb3086c79cbce6f9bd/lib/workers/repository/init/merge.ts#L50-L70

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/workers/repository/init/merge.ts [hunk 1]
```diff
@@ -51,7 +51,12 @@ export async function detectRepoFileConfig(): Promise<RepoFileConfig> {
   const cache = getCache();
   let { configFileName } = cache;
   if (configFileName) {
-    const configFileRaw = await platform.getRawFile(configFileName);
+    let configFileRaw: string | null;
+    try {
+      configFileRaw = await platform.getRawFile(configFileName);
+    } catch (err) {
+      configFileRaw = null;
```
Comment (viceice) [inline lib/workers/repository/init/merge.ts:58]: i think we should at least log the message for trouble shouting

**Review by viceice — COMMENTED** [review]
needs coverage

File: lib/workers/repository/init/merge.ts (hunk 1 above)
Comment (RahulGautamSingh) [reply lib/workers/repository/init/merge.ts:58]: On line 69 we handle the logging for undefined/null configFileRaw.

Comment (viceice) [reply lib/workers/repository/init/merge.ts:58]: but we should at least throw external host errors

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**